### ISSUE TITLE
feat: investigate and validate sample-specific VNTR length estimation

### ISSUE THREAD

**Issue by berntpopp** [hassansaei/VNtyper#332]
## Problem

VNtyper currently emits coverage measurements, but does not estimate the sample's MUC1 VNTR length or total repeat count. Track the investigation, validation, and eventual integration of a sample-specific estimator here.

The existing length fields describe reference geometry and must not be interpreted as sample allele lengths.

## Current implementation

`vntyper/scripts/coverage_stats.py::summarise_coverage` computes:

- `vntr_array_length`: the number of reference-array positions measured.
- `vntr_array_depth_sum`: summed depth across those positions.
- `vntr_array_depth_sum_per_unit_length`: array depth sum divided by the fixed `DEPTH_SUM_REFERENCE_LENGTH = 3481`; this is a coverage measurement, not an estimated repeat count.
- `vntr_flank_mean_depth`: mean depth over unique flanks, normally 190 bp per side.

The coverage stage is wired into `vntyper/scripts/pipeline.py`, writes `coverage_summary.tsv`, and exposes these measurements in the HTML report. Related groundwork: #222.

## Proposed scope

1. Define the estimand and output units explicitly: total diploid repeat count, mean allele length, or individual allele lengths. Do not imply that a total identifies the longer allele.
2. Evaluate candidate methods using independently measured sample lengths. Assess whether existing coverage measurements provide sufficient information or additional measurements are required.
3. Validate calibration and error across reference assemblies, coverage levels, capture assays, and alignment/counting policies. Simulation may support development, but real-sample validation is required before user-facing estimates.
4. Keep estimator calculations in a focused, independently testable module. Preserve current coverage fields and genotype decisions.
5. Once validation supports integration, add an explicit estimate, method/calibration provenance, and assessability information to structured output and reports. Represent unavailable estimates explicitly rather than as zero.
6. Treat any long-VNTR warning as a separately validated decision with a documented threshold and config-driven wording. An estimate alone does not establish a warning threshold.

## Calibration workflow and dependency

Use #269 (`vntyper calibrate`) to fit and validate this estimator from samples with independently measured VNTR lengths and corresponding BAM/CRAM or paired FASTQ data. Completed VNtyper runs may be reused when they contain all required estimator measurements and provenance.

Ownership:

- **This issue (#332):** define the estimator, measurement/feature schema, prediction target and units; extract the required measurements; consume a compatible calibration artifact at runtime; expose estimates and assessability in outputs/reports.
- **#269:** pair length truth with read data/features, fit the model, evaluate held-out performance, and export a versioned calibration artifact with parameters, provenance, and applicability conditions.

The shared contract must distinguish per-allele lengths from diploid totals, retain truth uncertainty/censoring, pin counting and feature semantics, and prevent repeated datasets from the same sample leaking between training and validation. Length-only calibration must not require carrier-status labels. The existing coverage metrics are candidate inputs; they do not by themselves establish a calibrated estimator.

Integration depends on a validated model and the artifact contract from #269. Any warning threshold remains a separate calibrated decision.

### Implementation order

1. **Shared intake in #269:** bind sample truth to input/read artifacts, resolve explicit aliases, audit duplicates, and freeze specimen/family groups before defining training and validation partitions.
2. **Caller calibration in #269:** integrate adVNTR capture and its existing background fitter as described there. This is the first caller-calibration deliverable; length-feature research can proceed independently against the same intake contract.
3. **Measurement-only work in this issue:** add pure length-feature calculations and an explicit feature schema, without emitting a production length or changing genotype decisions.
4. **Calibration and validation in #269:** compare candidate models against a training-only constant baseline; export a compatible model only after a documented validation decision.
5. **Inference/report integration here:** load that frozen artifact, check applicability, and emit an estimate with provenance and assessability. Keep a long-VNTR warning as a separately evaluated decision.

### First target and features

The initial target is **total diploid repeat count**, `N_total = N_short + N_long`, with truth supplied in repeat units. This target does not identify either allele individually. If truth arrives in bp, require an explicit definition/conversion policy rather than silently dividing every sequence length by a nominal repeat size.

Obtain per-base aligned depth with a pinned counting policy. The initial candidate policy is `samtools depth -a -q 0 -Q 0`: include zero-depth positions and MAPQ-zero alignments, and record samtools' other filtering and overlapping-mate semantics. Treat overlap removal (`-s`) as a distinct policy requiring its own evaluation. Distinguish absent measurements from evaluated zero depth.

Evaluate these candidate features:

- **Core/invariant ratio:** `A = mean_depth(core positions) / mean_depth(invariant end-repeat positions)`. Use explicit assembly-specific interval classes, not motif-name membership: identical motif names can occur in different structural contexts. A versioned, validated annotation is a prerequisite; do not assume one build's annotation applies to another.
- **Array/flank ratio:** `F = mean_depth(reference array) / mean_depth(unique flanks)`. Reuse existing array geometry, depth sums and flank depth where their counting policy matches. The existing reference-normalized depth column is still a coverage measurement, not a sample repeat count.

For comparison only, the physical assumptions imply:

```text
A physical prior: N_total = 2 * R_core * A + 2 * R_invariant
F physical prior: N_total = 2 * (L_reference_array / unit_length) * F
```

Here `R_core` and `R_invariant` are annotated reference repeat-copy counts, `L_reference_array` is the reference array length in bp, and `unit_length` is the declared repeat-unit length in bp. These priors assume comparable read recovery across the numerator and denominator; A additionally assumes invariant end-repeat copy number. Derive constants from the declared geometry/annotation. They are candidate assumptions to test, not calibrated defaults.

### Calibration algorithm

Start with deliberately small models: `N_hat = intercept + slope * A` and the analogous model for F. A joint A/F model may be an explicitly exploratory comparison, with collinearity and coefficient-stability checks. Compare all candidates with a baseline predicting the training set's mean repeat count.

For each held-out group: fit preprocessing and coefficients on training data only, predict the held-out samples, then aggregate errors. Use specimen/family grouping where available; technical replicates and alternate alignments of the same sample must never cross folds. Record unknown grouping as a validation limitation. Use nested selection if choosing features/models/hyperparameters, and evaluate the frozen choice on an untouched validation set.

Report mean/median absolute error in repeat units, bias, RMSE, out-of-sample R2, and agreement within a declared tolerance. Include transfer across supported assay/assembly conditions and sensitivity to counting policy. Correlation or in-sample fit alone is not an acceptance criterion. Predefine the required improvement and acceptable error before evaluating the validation set; do not automatically promote the best of weak candidate models.

### Code and artifact boundaries

- Put interval validation and ratio calculations in a focused pure module (proposed `length_features.py`); pass depth arrays, annotation, and counting-policy identity explicitly.
- Keep BAM/depth I/O in the coverage/capture orchestration layer. Cache measurements only when input digest, geometry, tool and counting policy agree.
- Keep fitting/evaluation in #269. Put artifact validation and prediction in a separate pure module (proposed `length_estimation.py`).
- Version the feature/model schemas and record target, units, feature order, coefficients, annotation/geometry digest, counting policy, training provenance and applicability conditions. Keep fitted cohort artifacts and private manifests outside tracked source.
- Reject incompatible artifacts and invalid/non-finite inputs. Missing annotation, zero/insufficient normalizing depth, or unsupported calibration conditions must produce explicit unavailability/assessability handling, never a fabricated zero-length estimate.
- Proposed additive output fields: `estimated_total_repeat_count`, `length_estimation_status`, and `length_calibration_id`; finalize names with the structured-output contract before implementing them.

Use synthetic fixtures for tests: interval boundaries and zeros, motif name versus interval class, known-ratio arithmetic, missing/zero denominators, duplicate/alias handling, grouped split isolation, artifact compatibility, and training-only baseline/model predictions. No real sample identifiers or experimental results belong in committed fixtures or this issue.

## Acceptance criteria

- [ ] Estimand, units, assumptions, and counting policy are documented.
- [ ] The feature/calibration contract is shared with #269; a model fitted from paired length truth and read data can be loaded without hand-editing and reproduces its held-out predictions.
- [ ] Runtime validation rejects incompatible estimator schemas, units, or counting policies and handles unsupported calibration conditions explicitly.
- [ ] Validation records sample provenance, independent truth, error/bias, and relevant subgroup performance; limitations and unresolved calibration issues are explicit.
- [ ] A documented go/no-go decision precedes user-facing integration.
- [ ] If integrated, unit tests cover numerical correctness, missing geometry/data, zero or insufficient normalizing depth, and invalid/non-finite inputs.
- [ ] Output/report tests distinguish reference dimensions from sample estimates and verify unavailable-result handling.
- [ ] Existing coverage and genotype behavior remains compatible; no unsupported allele-specific or warning interpretation is introduced.

If validation is insufficient, retain the work as experimental and record the remaining evidence needed before integration.
